File: 7-kyu/going-to-the-cinema/c++/solution.cpp
#include <cmath>

class Movie
{
public:
    static int movie(int card, int ticket, double perc);
};

int Movie::movie(int card, int ticket, double perc)
{
    int ret = 1;
    double card_price = static_cast<double>(card) + static_cast<double>(ticket) * perc;
    int total_ticket = ticket;
    while (!(std::ceil(card_price) < total_ticket)) {
        card_price += static_cast<double>(ticket) * std::pow(perc, ++ret);
        total_ticket += ticket;
    }
    return ret;
}

File: 7-kyu/going-to-the-cinema/python/solution.py
from math import ceil

def movie(card, ticket, perc):
    ret = 1
    card_price = card + ticket * perc
    total_ticket = ticket
    while not (ceil(card_price) < total_ticket):
        ret += 1
        card_price += ticket * (perc ** ret)
        total_ticket += ticket
    return ret